src/main/java/oracle/r2dbc/OracleR2dbcWarning.java

```diff
@@ -0,0 +1,81 @@
+package oracle.r2dbc;
+
+import io.r2dbc.spi.R2dbcException;
+import io.r2dbc.spi.Result;
+
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * <p>
+ * A subtype of the {@link Result.Segment} interface that provides information
+ * on warnings raised by Oracle Database.
+ * </p><p>
+ * When a SQL command results in a warning, Oracle R2DBC emits a {@link Result}
+ * with an {@code OracleR2dbcWarning} segment in addition to any other segments
+ * that resulted from the SQL command. For example, if a SQL {@code SELECT}
+ * command results in a warning, then an {@code OracleR2dbcWarning} segment is
+ * included with the result, along with any {@link Result.RowSegment}s returned
+ * by the {@code SELECT}.
+ * </p><p>
+ * R2DBC drivers typically emit {@code onError} signals for {@code Message}
+ * segments that are not consumed by {@link Result#filter(Predicate)} or
+ * {@link Result#flatMap(Function)}. Oracle R2DBC does not apply this behavior
+ * for warning messages. If an {@code OracleR2dbcWarning}
+ * segment is not consumed by the {@code filter} or {@code flatMap} methods of
+ * a {@code Result}, then the warning is discarded and the result may be
+ * consumed as normal with with the {@code map} or {@code getRowsUpdated}
+ * methods.
+ * </p><p>
+ * Warning messages may be consumed with {@link Result#flatMap(Function)}:
+ * </p><pre>{@code
+ * result.flatMap(segment -> {
+ *   if (segment instanceof OracleR2dbcWarning) {
+ *     logWarning(((OracleR2dbcWarning)segment).getMessage());
+ *     return emptyPublisher();
+ *   }
+ *   else {
+ *     ... handle other segment types ...
+ *   }
+ * })
+ * }</pre><p>
+ * A {@code flatMap} function may also be used to convert a warning into an
+ * {@code onError} signal:
+ * </p><pre>{@code
+ * result.flatMap(segment -> {
+ *   if (segment instanceof OracleR2dbcWarning) {
+ *     return errorPublisher(((OracleR2dbcWarning)segment).warning());
+ *   }
+ *   else {
+ *     ... handle other segment types ...
+ *   }
+ * })
+ * }</pre>
+ * @since 1.1.0
+ */
+public interface OracleR2dbcWarning extends Result.Segment {
```

Comment:
Extending `Result.Message` would likely make more sense to pick up what's already provided by the spec.

Reply:
I think that emitting warnings as errors creates a lot of inconvenience in the first place. 

> Users are telling me that they don't want warnings to be emitted as onError signals.

I also understand the desire to inform users that something unexpected has happened that didn't immediately lead to failure. Most other drivers do not emit warnings as errors but rather allow consumption of warning (message) segments when users are interested in such detail.

Reply:
Thanks for clarifying on this; Knowing that other drivers handle it this way is very helpful!
I've made the suggested changes.